server/middleware/auth.go

```diff
@@ -0,0 +1,85 @@
+package middleware
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/medibloc/panacea-dep-consumer/panacea"
+	log "github.com/sirupsen/logrus"
+)
+
+type JwtAuthMiddleware struct {
+	panaceaGRPCClient panacea.GRPCClient
+}
+
+func NewJWTAuthMiddleware(grpcClient panacea.GRPCClient) *JwtAuthMiddleware {
+	return &JwtAuthMiddleware{
+		panaceaGRPCClient: grpcClient,
+	}
+}
+
+func (mw *JwtAuthMiddleware) Middleware(next http.Handler) http.Handler {
```

Comment:
The expiry of JWT probably should be short.
But what if the JWT is exposed/hijacked to others? Do we need something like nonce to prevent from replay attack as well?

Reply:
Yes, I think this is also need to add a nonce as account JWT in panacea-oracle.

Reply:
Let's not adopt the challenge-response mechanism in this PR. Let's merge this PR with minimum implementations. After that, we can enhance security.